### DISCUSSION TITLE
Alerts query does not fetch any data

### DISCUSSION THREAD

**Discussion by CostinSM** [grafana/grafana-csv-datasource#184] [Q&A]
Hello,

We have been trying to configure the newly redesigned alerts to work but with no success.
We are using the CSV data source to fetch our data with no issues, but when it comes to doing the same thing in the alerts query, with the same configuration that works in the other panels, there is no data and an API error pops up in the upper right corner.

![image](https://user-images.githubusercontent.com/87074202/150758484-3e26f14e-bcc9-4a63-87a5-93b28c9e4a93.png)

I opened a ticket to Grafana Support as well but since this data source is a community plugin, they directed me over here. They did however show me what the logs are saying so I'll paste the reply and log snippet here, maybe it can be of any help:

"From what I found so far in the log, it seems to complain about the data return by the query.

t=2022-01-23T05:15:54+0000 lvl=eror msg="API error" logger=context userId=21 orgId=1 uname=pangidoanbutar error="Failed to evaluate queries and expressions: failed to execute conditions: input data must be a wide series but got type not (input refid)" remote_addr="35.191.11.197, 127.0.0.1"

I tried to adjust the time range from the Last 10 minutes to the Last 24 hours in the alerting page, but still showed no Data. 
When I inspected the Raw Data from the panel, it looks good from the last 24 hours so I am not really sure why it's showing No Data on the alerting page."

Did anyone else encounter this issue before?

Regards,
Costin

**Comment by marcusolsson** [comment]
Your query return a Time field and a String field, which is not a valid time series. Have you tried setting the type for the cpu load to Number?

**Reply by CostinSM** [reply]
Hello Marcus,

Thank you for taking a look at our issue.
Indeed, the type String was an oversight on my part and I was hoping this would do the trick, however it did not.

I tried again with the type set to Number and there is still no data returned by the query but this time the error returned is the one found in the log by Grafana Support the first time.

![Alert_v2](https://user-images.githubusercontent.com/87074202/158437585-a903eebb-8fcc-4443-b55f-2546e3699698.jpg)

I also attached a picture from the panel where we do the same query for the same data with the same settings as in the alert panel: 

![CPU query panel](https://user-images.githubusercontent.com/87074202/158437763-207375d3-3ccc-4eec-8a65-5a91921385ca.jpg)

Regards,
Costin

**Reply by marcusolsson** [reply]
I think it's because there are duplicate datestimes, which makes Grafana detect it as "long" format. We could look into adding an option to convert from long to wide, but it would need another release.

In the meantime, could you try to deduplicate the utcDateTime field and see if that works?

**Reply by CostinSM** [reply]
Hello Marcus,

Here is a small sample of the data being read by the Grafana:
![image](https://user-images.githubusercontent.com/87074202/158574332-23356f86-6da4-4f9a-a292-4e597a166b61.png)

The collection of the data is done every minute and then they are inserted into a DB with the timestamp of the run. There is almost no chance of duplicates being fed to Grafana.

Even if this would be the case though, wouldn't this affect the panel query as well, and not only the query from the alert? Is there a difference between them in this sense?

Regards,
Costin

**Reply by marcusolsson** [reply]
Most of the panels convert long data into wide before visualizing them. Seems like the alerting engine doesn't though. Is there any way you could paste a sample of the CSV you're using so that I could play around with it?

**Reply by CostinSM** [reply]
For sure! I added two sample files; one with the data from above and one with the data from another panel, with multiple fields.

Thanks!
Costin

[CPU.csv](https://github.com/marcusolsson/grafana-csv-datasource/files/8290039/CPU.csv)
[Heap.csv](https://github.com/marcusolsson/grafana-csv-datasource/files/8290040/Heap.csv)

**Reply by CostinSM** [reply]
Hello Marcus,

Did you get the chance to take a look at the samples I added? :D

Thanks!
Costin

**Reply by CostinSM** [reply]
Hello Marcus,

I figured what the issue with the alert query was.. It did not have anything to do with the data source or the incoming data but with the time parameters that were set up with global variables in the dashboard panels. I was unaware the alert panel does not support the same variables and looking back, I think this is something I should have mentioned in the description of the thread.

The problem now is that by removing the variables, the alert query would fetch ALL the available data from the data source every time, and only look at the set interval (let's say 4h) for values above the threshold. This is totally unacceptable and I was wondering if you have any insight or workaround for this new issue..

![image](https://user-images.githubusercontent.com/87074202/167633218-6bc7cc19-e2ea-450d-9ceb-c0d690e78d51.png)

Regards,
Costin
